src/main/java/seedu/multitasky/logic/parser/FindCommandParser.java

```diff
@@ -24,15 +27,70 @@ public FindCommand parse(String args) throws ParseException {
         String trimmedArgs = args.trim();
 
         if (args.trim().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                                                   FindCommand.MESSAGE_USAGE));
+        }
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(trimmedArgs,
+                                                                ParserUtil.toPrefixArray(FindCommand.VALID_PREFIXES));
+        ArrayList<String> prefixesPresent = argumentMultimap.getPresentPrefixes(FindCommand.VALID_PREFIXES);
+        if (!hasValidPrefixCombination(prefixesPresent)) {
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
                                                    FindCommand.MESSAGE_USAGE));
         }
+        Set<String> keywords;
+        Optional<String> optKeywords = argumentMultimap.getPreamble();
+        if (optKeywords.isPresent()) {
+            keywords = new HashSet<>(Arrays.asList(optKeywords.get().split("\\s+")));
+        } else {
+            keywords = new HashSet<>();
+        }
+        Calendar startDate = getStartDate(argumentMultimap);
+        Calendar endDate = getEndDate(argumentMultimap);
+
+        return new FindCommand(keywords, startDate, endDate, prefixesPresent);
+    }
+
+    private Calendar getStartDate(ArgumentMultimap argumentMultimap) {
+        ArrayList<String> dateArgs = (ArrayList<String>) argumentMultimap.getAllValues(CliSyntax.PREFIX_FROM);
```

Comment:
Instead of using getAllValues, can we be consistent and stick with getValue(PREFIX_FROM) to get the rawDate args instead? so instead of the 2 lines, use return getDate(argMultimap.get(CliSyntax.PREFIX_FROM));
the difference : 
right now , your algorithm will search through all flags with from and concatenate them. 
e.g `find from president from day` will feed in `president day` to prettyTime

the algorithm with all the rest of the commands : 
only takes the very last flagged args as valid args
e.g `find from mistyped prseidnet day from president day` will feed in president day to prettyTime